ibm/service/satellite/data_source_ibm_satellite_storage_configuration.go

```diff
@@ -0,0 +1,122 @@
+
+package satellite
+
+import (
+	"fmt"
+
+	"github.com/IBM-Cloud/container-services-go-sdk/kubernetesserviceapiv1"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DataSourceIBMSatelliteStorageConfiguration() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceIBMSatelliteStorageConfigurationRead,
+
+		Schema: map[string]*schema.Schema{
+			"location": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The Location Name.",
+			},
+			"config_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the Storage Configuration.",
+			},
+			"config_version": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Version of the Storage Configuration.",
+			},
+			"storage_template_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The Storage Template Name.",
+			},
+			"storage_template_version": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The Storage Template Version.",
+			},
+			"user_config_parameters": {
+				Type:        schema.TypeMap,
+				Computed:    true,
+				Description: "The storage configuration parameters depending on the storage template.",
+			},
+			"user_secret_parameters": {
+				Type:      schema.TypeMap,
+				Sensitive: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Required:    true,
+				Description: "User Secret Parameters to pass as a Map of string key-value.",
+			},
+			"storage_class_parameters": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeMap,
+					Elem: &schema.Schema{
+						Type: schema.TypeString,
+					},
+					Description: "The List of Storage Class Parameters as a list of a  Map of string key-value.",
+				},
+			},
+			"uuid": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The Universally Unique IDentifier (UUID) of the Storage Configuration.",
+			},
+		},
+	}
+}
+
+func dataSourceIBMSatelliteStorageConfigurationRead(d *schema.ResourceData, meta interface{}) error {
+	satClient, err := meta.(conns.ClientSession).SatelliteClientSession()
+	if err != nil {
+		return err
+	}
+
+	satLocation := d.Get("location").(string)
+	d.Set("location", satLocation)
+	getSatelliteLocation := &kubernetesserviceapiv1.GetSatelliteLocationOptions{
+		Controller: &satLocation,
+	}
+	_, _, err = satClient.GetSatelliteLocation(getSatelliteLocation)
+	if err != nil {
+		return fmt.Errorf("[ERROR] Location not found! - %v", err)
+	}
+
+	storageConfigName := d.Get("config_name").(string)
+	getStorageConfigurationOptions := &kubernetesserviceapiv1.GetStorageConfigurationOptions{
+		Name: &storageConfigName,
+	}
+	result, _, err := satClient.GetStorageConfiguration(getStorageConfigurationOptions)
+	if err != nil {
+		return err
```

Comment:
As per the code I see location and config_name only as inputes
Remaining all shd be computed